### PR TITLE
EVG-12903 allow task to be not reachable for git tag builds

### PR DESCRIPTION
**Pull request by ablack12**
like patchable: false but "git-taggable: false" (i didn't like git taggable though)

### CODE REVIEW

File: validator/project_validator.go [hunk 1]
```diff
@@ -1570,7 +1577,7 @@ func dependencyMustRun(target model.TVPair, current model.TVPair, depReqs depend
 	if depReqs.requireOnNonPatches && (taskUnit.SkipOnNonPatchBuild() || taskUnit.SkipOnNonGitTagBuild()) {
 		return false, nil
 	}
-	if depReqs.requireOnGitTag && (taskUnit.SkipOnNonPatchBuild()) {
+	if depReqs.requireOnGitTag && (taskUnit.SkipOnNonPatchBuild() || taskUnit.SkipOnGitTagBuild()) {
```
Comment (Kimchelly) [inline validator/project_validator.go:1580]: On L1521, it seems like `requireOnGitTag` should also require that `!SkipOnGitTagBuild()`.

File: validator/project_validator_test.go [hunk 2]
```diff
@@ -3284,6 +3302,28 @@ func TestDependencyMustRun(t *testing.T) {
...
+					Patchable: falsePtr,
+					DependsOn: []model.TaskUnitDependency{
+						{Name: "B", Variant: "ubuntu"},
+					},
+				},
+				{TaskName: "B", Variant: "ubuntu"}: {
+					AllowForGitTag: falsePtr,
+				},
+			},
+			expectDependencyFound: false,
```
Comment (Kimchelly) [inline validator/project_validator_test.go:3325]: Can we test for the case of `requireOnGitTag` is true and `AllowForGitTag` is also true?

**Review by Kimchelly — CHANGES_REQUESTED** [review]

**Review by john-m-liu — APPROVED** [review]

File: validator/project_validator.go (hunk 1 above)
Comment (ablack12) [reply validator/project_validator.go:1580]: good catch!

File: validator/project_validator_test.go (hunk 2 above)
Comment (ablack12) [reply validator/project_validator_test.go:3325]: Done

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**